rpc-client/src/nonblocking/rpc_client.rs

```diff
@@ -2179,8 +2179,10 @@ impl RpcClient {
         &self,
         commitment: CommitmentConfig,
         max_stake_percent: f32,
+        timeout: Option<Duration>,
```

Comment:
My only concern is this is a public function in one of our more public crates. I don't expect many people to be using this, but it is technically an API breakage nonetheless.

The timeout is nice to have, but not a must if we want to avoid the messiness that comes with adjusting API

Reply:
We could add a new function `wait_for_max_stake_with_timeout()`.
And change this function body to be just
```rust
    await self.wait_for_max_stake_with_timeout(commitement, max_stake_percent, None)
```

Addition of a new function, technically, is also an API breaking change.
Due to the possibility that someone could have implemented a trait that introduces a function with the same name.
But it seems like a very unlikely issue here.

We could potentially also mark it with the `dev-context-only-utils` feature.
But, I think, a timeout is actually a useful functionality...

Reply:
Brennan pointed out that this function name is actually confusing, as it is not waiting for any of the nodes to reach the maximum specified stake level.
It actually waits for the maximum stake level to fall below the specified level.

In which case we could take the opportunity to rename and, possibly, even document, a new function.
Maybe something like
```rust
    /// Wait for the largest single stake holder stake percentage to fall below the specified
    /// percentage level.
    ///
    /// This function is mostly used when starting a new local chain, and we want to wait for the
    /// moment when the stake, initially full under the genesis node control, to be distributed
    /// among all the other nodes.  Common value for `max_stake_percent` is `0.33`, meaning that no
    /// single node can stop the chain any more.
    pub async fn wait_for_max_stake_below_with_timeout(
        &self,
        commitment: CommitmentConfig,
        max_stake_percent: f32,
        timeout: Option<Duration>,
    ) -> ClientResult<()> {
```

Reply:
Yeah, I'll add a secondary set of APIs that include timeout instead of modifying the existing ones. Thanks for catching this.